src/include/platform/FailSafeContext.h

```diff
@@ -55,12 +55,20 @@ class FailSafeContext
         return (accessingFabricIndex == mFabricIndex);
     }
 
-    inline bool NocCommandHasBeenInvoked() { return mNocCommandHasBeenInvoked; }
+    inline bool NocCommandHasBeenInvoked() { return mAddNocCommandHasBeenInvoked || mUpdateNocCommandHasBeenInvoked; }
+    inline bool AddNocCommandHasBeenInvoked() { return mAddNocCommandHasBeenInvoked; }
+    inline bool UpdateNocCommandHasBeenInvoked() { return mUpdateNocCommandHasBeenInvoked; }
 
-    inline void SetNocCommandInvoked(FabricIndex nocFabricIndex)
+    inline void SetAddNocCommandInvoked(FabricIndex nocFabricIndex)
     {
-        mNocCommandHasBeenInvoked = true;
-        mFabricIndex              = nocFabricIndex;
+        mAddNocCommandHasBeenInvoked = true;
+        mFabricIndex                 = nocFabricIndex;
+    }
+
+    inline void SetUpdateNocCommandInvoked(FabricIndex nocFabricIndex)
+    {
+        mUpdateNocCommandHasBeenInvoked = true;
+        mFabricIndex                    = nocFabricIndex;
     }
```

Comment:
We must validate that a AddNOC/UpdateNOC happening on a different fabric doesn't cause these methods to be called and override current fabric association. Did you audit the NOC cluster to make sure we properly keep track of fail-safe context in case we receive multiple AddNOC/UpdateNOC from many fabrics? There is supposed to only be a single one by fail-safe period but I don't see how this is verified in the NOC cluster code.

Reply:
Shall this be guaranteed by CommissioningWindowManager?     Currently, the Fabric Index associated with the armed fail-safe context is always updated to match the Fabric Index associated with the AddNOC/UpdateNOC command being invoked.

Reply:
This is enforced in the NOC cluster via the `if (!failSafeContext.IsFailSafeArmed(commandObj->GetAccessingFabricIndex()))` checks in AddNOC and UpdateNOC.